src/pyfmi/fmi_algorithm_drivers.py

```diff
@@ -324,33 +319,6 @@ def __init__(self,
 
             self.model.set(input_names, input_values)
 
-        if self.options["result_handling"] == "file":
```

Comment:
Moved this to separate function for readability and because it now requires to be set before the _set_options() call, since _set_options now checks result_handler.supports, rather than a specific ResultHandler instance.